packages/web/components/Dashboard/Settings/Subscription/SubscriptionForm.tsx

```diff
@@ -0,0 +1,165 @@
+import PaymentFormModal from './PaymentFormModal'
+
+type SubscriptionFormProps = {
+  user: UserType
+}
+
+const SubscriptionForm = ({ user }: SubscriptionFormProps) => {
+  const { t } = useTranslation('settings')
+  let subscriptionPlan: string | undefined
+  // (period) => string
+  if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Monthly) {
+    subscriptionPlan = '1 Month / £12'
+  } else if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Quarterly) {
+    subscriptionPlan = '3 Months / £30'
+  } else if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Annualy) {
+    subscriptionPlan = '1 Year / £100'
+  }
+  const isCancelling = user.membershipSubscription?.cancelAtPeriodEnd
+  const [showPaymentForm, setShowPaymentForm] = useState(!user.isPremiumUser)
+  const [showPaymentFormModal, setShowPaymentFormModal] = useState(false)
+
+  const [updateSubscriptionRenewal] = useUpdateSubscriptionRenewalMutation({
+    onCompleted: () => {
+      // TODO: bust the cache for the User
+      toast.success(t('You have cancelled your subscription'))
+    },
+    onError: () => {
+      toast.error(t('There was a problem cancelling your subscription'))
+    },
+  })
+
+  const [CancelSubscriptionConfirmationModal, confirmCancellation] = useConfirmationModal({
+    title: 'Cancel Your Journaly Premium Subscription',
+    body: 'Are you sure you want to cancel your subscription?'
+  })
+
+  const handleCancelSubscription = async () => {
+    if (!(await confirmCancellation())) return
+    
+    updateSubscriptionRenewal({
+      variables: {
+        cancelAtPeriodEnd: true,
+      }
+    })
+  }
+
+  const SubscriptionStatusBadge = () => {
+    return (
+      <>
+        <span className="badge">
+          {user.isPremiumUser ? 'premium user' : 'free user'}
+        </span>
+        <style jsx>{`
+          .badge {
+            padding: 10px;
+            border-radius: 5px;
+            text-transform: uppercase;
+            font-size: ${theme.typography.paragraphSM}
+            font-weight: 600;
+            background-color: ${!user.isPremiumUser ? theme.colors.gray100 : theme.colors.greenLight};
+            color: ${!user.isPremiumUser ? theme.colors.gray600 : theme.colors.greenDark};
+          }
+        `}</style>
+      </>
+    )
+  }
+
+  return (
+    <>
+      {showPaymentFormModal && (
+        <PaymentFormModal onClose={() => setShowPaymentFormModal(false)} />
+      )}
+      <div className="page-container">
+        <p style={{ marginBottom: '20px' }}>{t('subscription.copy')}</p>
+        <p style={{ marginBottom: '10px' }}><strong>Subscription status:</strong> <SubscriptionStatusBadge /></p>
+        {user.isPremiumUser && (
+          <>
+            <p><strong>Current Plan:</strong> {subscriptionPlan}</p>
+            {user.membershipSubscription?.lastFourCardNumbers && (
+              <CardOnFile last4="1234" onUpdateCard={() => {}} />
```

Comment:
Are we doing the update payment method flow as a followup?

Reply:
No I've actually already written the mutation, just forgot that this placeholder was here hehe, will add it in